corehq/apps/locations/permissions.py

```diff
@@ -98,3 +102,22 @@ def _inner(request, domain, *args, **kwargs):
             return view_fn(request, domain, *args, **kwargs)
         raise Http404()
     return locations_access_required(_inner)
+
+
+def can_edit_form_location(domain, user, instance_id):
+    if not toggles.RESTRICT_FORM_EDIT_BY_LOCATION.enabled(domain):
+        return True
+
+    location = get_xform_location(XFormInstance.get(instance_id))
```

Comment:
this should probably use `get_document_or_404` to avoid crashing if you specify unknown / bad doc ids